opentreemap/treemap/models.py

```diff
@@ -695,7 +695,21 @@ def cast_to_subtype(self):
         Return the concrete subclass instance. For example, if self is
         a MapFeature with subtype Plot, return self.plot
         """
-        return getattr(self, self.feature_type.lower())
+        ft = self.feature_type
+        if hasattr(self, ft.lower()):
+            return getattr(self, ft.lower())
+        else:
+            # a concession: we make an extra query because instances
+            # don't know how to cast along a heirarchy, but classes
+            # know how fetch subtype and a query easily gets an
+            # instance of the subtype.  This is a single row query not
+            # likely to happen as part of a list-loop, and will only
+            # happen for features that are not *direct* descendents of
+            # MapFeature.
+            return (self
+                    .__class__
+                    .get_subclass(ft)
+                    .objects.get(pk=self.id))
```

Comment:
Fix pushed. I dropped the non-exhaustive elif clause because it would have produced a `DoesNotExist` error the same as attempting to access it directly.